### PR TITLE
Replace a couple of long by intptr_t

### PR DESCRIPTION
**Pull request by bellenot**
Replace casting of pointers to long that cannot work on Windows 64 bit (where pointers are 64 bit and long 32 bit)

### CODE REVIEW

File: core/metacling/src/TClingBaseClassInfo.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ class TClingBaseClassInfo {
    bool          IsValid() const;
    int           Next();
    int           Next(int onlyDirect);
-   ptrdiff_t     Offset(void * address = 0, bool isDerivedObject = true) const;
+   intptr_t      Offset(void * address = 0, bool isDerivedObject = true) const;
```
Comment (amadio) [inline core/metacling/src/TClingBaseClassInfo.h:75]: Why is this one needed? I thought `intptr_t` and `ptrdiff_t` would mostly be equivalent. Semantics?

Comment (bellenot) [reply core/metacling/src/TClingBaseClassInfo.h:75]: Yes, semantics. I use `intptr_t` almost everywhere

Comment (pcanal) [reply core/metacling/src/TClingBaseClassInfo.h:75]: Yes, but semantically here this is an `offset` and hence `ptrdiff_t`, isn't it?

Comment (bellenot) [reply core/metacling/src/TClingBaseClassInfo.h:75]: Right. Do you want me to change it?

Comment (pcanal) [reply core/metacling/src/TClingBaseClassInfo.h:75]: Yes, please :)

Comment (bellenot) [reply core/metacling/src/TClingBaseClassInfo.h:75]: OK

Comment (pcanal) [reply core/metacling/src/TClingBaseClassInfo.h:75]: >  result of pointer calculation?

When it is a difference, yes.  So the return value of the wrapper, 'fOffset` and `fIterStack` should be using `ptrdiff_t`

**Comment by bellenot** [comment]
BTW, one should really stop casting pointers to long. I was surprised to see this again, we are in 2022!

**Review by pcanal — APPROVED** [review]
Thanks.

**Pull request merged**